Diff/diff.py

```diff
@@ -170,26 +214,41 @@ def run(self, edit, **kwargs):
         )
 
         try:
-            common_path_length = len(os.path.commonpath(view_names))
+            common_path_length: int = len(os.path.commonpath(view_names))
             if common_path_length <= 1:
                 common_path_length = 0
             else:
                 common_path_length += 1
         except ValueError:
             common_path_length = 0
         view_names = list(map(lambda name: name[common_path_length:], view_names))
-        show_diff_output(diff, views[0], views[0].window(), f'{view_names[0]} -> {view_names[1]}', 'diff_views', 'diff_tabs_to_buffer')
-
-    def is_enabled(self, **kwargs):
+        view: sublime.View = views[0]
+        window: typing.Optional[sublime.Window] = view.window()
+        if window is not None:
+            show_diff_output(
+                diff,
+                view,
+                window,
+                f'{view_names[0]} -> {view_names[1]}',
+                'diff_views',
+                'diff_tabs_to_buffer'
+            )
+
+    def is_enabled(self, **kwargs) -> bool:
         return self.is_visible(**kwargs)
 
-    def is_visible(self, **kwargs):
+    def is_visible(self, **kwargs) -> bool:
         views = get_views_from_tab_context(self.view, **kwargs)
-        return len(views) == 2
-
-    def description(self, **kwargs):
-        selected_views = list(get_selected_views(self.view.window()))
-        if len(selected_views) == 2:
-            return 'Diff Selected Tabs...'
+        if views is None:
+            return False
+        else:
+            return len(views) == 2
+
+    def description(self, **kwargs) -> str:
+        window: typing.Optional[sublime.Window] = self.view.window()
+        if window is not None:
+            selected_views = list(get_selected_views(window))
```

Comment:
Maybe just return empty string on `window is None` to make the logic read better.